### PR TITLE
ROD-92 Create a reference number page

### PR DESCRIPTION
**Pull request by MikeMaratheftis**
## What? 
[ROD-92](https://collaboration.homeoffice.gov.uk/jira/browse/ROD-92) - Create a reference number page - ROD main flow

## Why? 
Allows user to enter a reference number

## How? 
- Added reference number page
- Added custom validations for page

## Testing?
Tested on local machine 

## Check list

- [x] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [ ] I have written tests (if relevant)
- [x] I have created a JIRA number for my branch
- [x] I have created a JIRA number for my commit
- [x] I have followed the chris beams method for my commit https://cbea.ms/git-commit/
here is an [example commit](https://github.com/UKHomeOfficeForms/hof/commit/810959f391187c7c4af6db262bcd143b50093a6e)
- [x] Ensure drone builds are green especially tests
- [x] I will squash the commits before merging

### CODE REVIEW

**Comment by RobertMcCann** [comment]
update next step for /reference-number to be /your-documents as per AC5 in ticket

the check your answers page isn't showing up as per the figma, needs a field that displays the type of reference numbers the user has selected (can be all of them)

**Comment by RobertMcCann** [comment]
For all of your field names that you have added to this PR I would prefix them with 'rod' so it's easier to read, search for in IDE and some of these field names are already in use in other journeys e.g. enter-case-id is in CNC 

So I would change all these fields to 'rod-case-id' etc and update everywhere else they appear, you can use VSC change all occurrences so that nothing gets missed

File: apps/rod/sections/summary-data-sections.js [hunk 1]
```diff
@@ -60,6 +60,26 @@ module.exports = {
     {
       step: '/main-applicant',
       field: 'main-applicant-nationality'
+    },
+    {
+      step: '/reference-number',
+      field: 'enter-reference-number'
```
Comment (vinodhasamiyappanHO) [inline apps/rod/sections/summary-data-sections.js:66]: field name is different from the pages.json entry for label. It should be `reference-number`.

And if more than one option is selected, the value needs to be parsed and should display in the summary page as like below.

```
Case ID
Home Office reference number
Payment reference number
Courier reference number
```

File: apps/rod/behaviours/custom-validations.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const validators = require('hof/controller/validation/validators');
+
+module.exports = superclass =>
+  class extends superclass {
+    validateField(key, req) {
+      const validationErrorFunc = (type, args) =>
+        new this.ValidationError(key, { type: type, arguments: [args] });
+        // eslint-disable-next-line max-len
+      if (req.form.values['rod-reference-number'].includes('ho-reference-number') && key === 'rod-ho-reference-number') {
```
Comment (vivekkumar-ho) [inline apps/rod/behaviours/custom-validations.js:9]: we can move the if condition to multiline so don't have to disable the rule

Comment (MikeMaratheftis) [reply apps/rod/behaviours/custom-validations.js]: Done :)

**Review by vivekkumar-ho — CHANGES_REQUESTED** [review]
`apps/rod/behaviours/custom-validations.js` already exists. Probably better to use that file for all the custom validations and delete the additional file.

File: apps/rod/translations/src/en/validation.json [hunk 3]
```diff
@@ -80,5 +71,42 @@
...
+    },
+    "rod-document-type": {
+        "required": "Enter the other type of document you need us to return",
+        "notUrl": "Answer must not contain a URL. Remove any links, URLs or special characters like /@:%_+~#=\\ from your answer",
+        "maxlength": "Other document type must be less than 100 characters"
+    },
+    "document-description": {
+        "notUrl": "Answer must not contain a URL. Remove any links, URLs or special characters like /@:%_+~#=\\ from your answer",
+        "maxlength": "Document descriptions cannot be more than than 5,000 characters"
+    },
```
Comment (vivekkumar-ho) [inline apps/rod/translations/src/en/validation.json:83]: It seems there are updated accidentally ?

**Review by vivekkumar-ho — COMMENTED** [review]
validation files are now merged

**Review by vivekkumar-ho — APPROVED** [review]

**Pull request merged**